File: 프로그래머스/unrated/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.java
//1. 일반 풀이
// class Solution {
//     public int solution(int[][] board, int k) {
//         int answer = 0;
//         for(int i = 0; i <= Math.min(board.length-1,k); i++){
//             for(int j = 0; j <= Math.min(board[i].length-1,k); j++){
//               if(i+j<=k) answer += board[i][j]; 
//             }
//         }
//         return answer;
//     }
// }

//2. 스트림 풀이
import java.util.stream.IntStream;
class Solution {
    public int solution(int[][] board, int k) {
        return IntStream.range(0, board.length).flatMap(i -> IntStream.range(0, board[i].length).filter(j -> i+j <= k).map(j -> board[i][j])).sum();
}
}

File: 프로그래머스/0/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.cpp
#include <string>
#include <vector>
#include <cmath>
using namespace std;

int solution(vector<vector<int>> board, int k) {
    int answer = 0;
    for(int i = 0; i < board.size(); i++){
        for(int j = 0; j < board[0].size(); j++){
            if(i + j <= k){
                answer += board[i][j];
            }else{
                break;
            }
        }
    }
    return answer;
}